src/smpi/smpi_replay.cpp

```diff
@@ -998,44 +1223,76 @@ void smpi_replay_run(int *argc, char***argv){
     XBT_DEBUG("Force context switch by smpi_execute_flops  - Sleeping for 0.0 flops ");
     smpi_execute_flops(0.0);
   }
+}
 
-  /* Actually run the replay */
-  xbt_replay_action_runner(*argc, *argv);
 
+/*
+ * This function only initializes smpi_replay. It does not start the replay.
+ * You should call this one if you want to be able to register new actions
+ * from your code.
+ * To really start the replay you must call smpi_replay_run.
+ */
+void smpi_replay_init(int *argc, char ***argv)
+{
+  smpi_replay_initialize(argc, argv);
+}
+
+/*
+ * This function starts the replay of the traces. Your custom actions must be
+ * registered before calling this function.  It was previously calles
+ * smpi_replay_run, but I had to rename it because someone used this name on
+ * original SimGrid master branch.
+ */
+void smpi_replay_start(int *argc, char ***argv)
+{
+  xbt_replay_action_runner(*argc, *argv);
+ 
   /* and now, finalize everything */
   double sim_time= 1.;
   /* One active process will stop. Decrease the counter*/
-  XBT_DEBUG("There are %zu elements in reqq[*]", get_reqq_self()->size());
```

Comment:
My code is actually older than your conversion to std::. When I wrote this code this variable was a xbt_dynar_t, which I replaced with a xbt_dict_t, so I could match requests inside MPI_Wait.

But, I will look to see if there is a dictionary type I can use in std::. This may take some time, since I know very little C++.